packages/ui/src/hooks/usePendingTx.tsx

```diff
@@ -1,39 +1,169 @@
 import { useCallback, useEffect, useMemo, useState } from 'react'
-import { useApi } from '../contexts/ApiContext'
-import { MultiProxy } from '../contexts/MultiProxyContext'
-import { MultisigStorageInfo } from '../types'
+import { useMultiProxy } from '../contexts/MultiProxyContext'
+import { HexString, MultisigStorageInfo } from '../types'
 import { useMultisigCallSubscription } from './useMultisigCallsSubscription'
 import { isEmptyArray } from '../utils'
 import { useAccountId } from './useAccountId'
+import { ApiPromise } from '@polkadot/api'
+import { useApi } from '../contexts/ApiContext'
+import { ISanitizedCall, getDisplayArgs, isProxyCall, parseGenericCall } from '../utils'
+import { GenericCall } from '@polkadot/types'
+import { AnyJson, AnyTuple } from '@polkadot/types/types'
+import dayjs from 'dayjs'
+import localizedFormat from 'dayjs/plugin/localizedFormat'
+dayjs.extend(localizedFormat)
 
 export interface PendingTx {
   from: string
   hash: string
   info: MultisigStorageInfo
 }
-export const usePendingTx = (multiProxy?: MultiProxy) => {
+
+export interface CallDataInfoFromChain {
+  callData?: HexString
+  hash?: string
+  name?: string
+  args?: AnyJson
+  info?: PendingTx['info']
+  from: string
+  timestamp?: Date
+  multiProxyAddress?: string
+}
+
+type AggGroupedByDate = { [index: string]: CallDataInfoFromChain[] }
+
+export const getMultisigInfo = (c: ISanitizedCall): Partial<CallDataInfoFromChain>[] => {
+  const result: Partial<CallDataInfoFromChain>[] = []
+
+  const getCallResult = (c: ISanitizedCall) => {
```

Comment:
is it some golang practice to call a param with single letter ? :)

Reply:
I think this is not specific to a language.. made this more readable in [ee33cfd](https://github.com/ChainSafe/Multix/pull/434/commits/ee33cfd0fa73946114ee94ba7ee4b5a04283526f)